### PR TITLE
Currency: Added build_script, build_info.json and LICENSE for torchvision

### PR DESCRIPTION
**Pull request by Shafi-Hussain**
…sion

## Checklist

- [ ] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [ ] Have you validated script on UBI 9 container
- [ ] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

File: t/torchvision/torchvision_ubi_9.3.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+OS_NAME=$(cat /etc/os-release | grep ^PRETTY_NAME | cut -d= -f2)
+PYTHON_VER=${2:-3.9}
+
+WORKDIR=$(pwd)
+
+dnf install -y git jq
+
+# install dependency - pytorch (skip tests)
+PYTORCH_VERSION=${3:-$(curl -sSL https://api.github.com/repos/pytorch/pytorch/releases/latest | jq -r .tag_name)}
+curl -sL https://raw.githubusercontent.com/ppc64le/build-scripts/master/p/pytorch/pytorch_ubi_9.3.sh | sed '/pip install pytest/q' | bash -s $PYTORCH_VERSION $PYTHON_VER
```
Comment (shubham-dayma-ibm) [inline t/torchvision/torchvision_ubi_9.3.sh:34]: We should be having instructions to install `pytorch` here in this build script itself, rather downloading the build script.

File: t/torchvision/torchvision_ubi_9.3.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+PILLOW_VERSION=${4:-$(curl -sSL https://api.github.com/repos/python-pillow/Pillow/releases/latest | jq -r .tag_name)}
+curl -sL https://raw.githubusercontent.com/ppc64le/build-scripts/master/p/pillow/pillow_v11.0.0_ubi_9.3.sh | sed '/# Run tests to verify installation/q' | bash -s $PILLOW_VERSION $PYTHON_VER
+
+cd $WORKDIR
+
+git clone $PACKAGE_URL
+cd $PACKAGE_NAME
+git checkout $PACKAGE_VERSION
+
+if ! (MAX_JOBS=$(nproc) python setup.py bdist_wheel && pip install dist/*.whl); then
```
Comment (shubham-dayma-ibm) [inline t/torchvision/torchvision_ubi_9.3.sh:57]: Build-script should only have steps to install package and not building the package. Consider using `pip install .` to install package from source or similar. Do not build wheel in build-script.

**Review by shubham-dayma-ibm — CHANGES_REQUESTED** [review]

File: t/torchvision/torchvision_ubi_9.3.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+#                     package and/or distribution. In such case, please
+#                     contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+
+PACKAGE_NAME=vision
+PACKAGE_VERSION=${1:-v0.16.1}
+PACKAGE_URL=https://github.com/pytorch/vision.git
+OS_NAME=$(cat /etc/os-release | grep ^PRETTY_NAME | cut -d= -f2)
+PYTHON_VER=${2:-3.9}
```
Comment (shubham-dayma-ibm) [inline t/torchvision/torchvision_ubi_9.3.sh:26]: Remove `PYTHON_VER` variable and python installation instructions

File: t/torchvision/torchvision_ubi_9.3.sh [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            libpng-devel freetype-devel lcms2-devel libwebp-devel tcl-devel tk-devel \
+            harfbuzz-devel fribidi-devel libraqm-devel libimagequant-devel libxcb-devel
+
+if ! command -v python; then
+    ln -s $(command -v python${PYTHON_VER}) /usr/bin/python
+fi
+if ! command -v pip; then
+    ln -s $(command -v pip${PYTHON_VER}) /usr/bin/pip
+fi
+
```
Comment (shubham-dayma-ibm) [inline t/torchvision/torchvision_ubi_9.3.sh:48]: No need to create symbolic link for `python`.
Your build script should only have the instructions about the package requirements. 
Installation of python, supporting different python versions etc, is already taken care in the wrapper script. So we don't need repetitive instructions.
For testing purpose you can create python venv in your ssh terminal and hit this shell script.

File: t/torchvision/torchvision_ubi_9.3.sh [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+WORKDIR=$(pwd)
+
+dnf install -y https://mirror.stream.centos.org/9-stream/BaseOS/ppc64le/os/Packages/centos-gpg-keys-9.0-24.el9.noarch.rpm \
+            https://mirror.stream.centos.org/9-stream/BaseOS/`arch`/os/Packages/centos-stream-repos-9.0-24.el9.noarch.rpm \
+            https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
+dnf config-manager --add-repo https://mirror.stream.centos.org/9-stream/AppStream/`arch`/os
+dnf config-manager --set-enabled crb
+dnf install -y git cmake ninja-build g++ rust cargo jq \
+            python${PYTHON_VER}-devel python${PYTHON_VER}-wheel python${PYTHON_VER}-pip python${PYTHON_VER}-setuptools \
```
Comment (shubham-dayma-ibm) [inline t/torchvision/torchvision_ubi_9.3.sh:37]: Remove python installation instructions

**Review by shubham-dayma-ibm — CHANGES_REQUESTED** [review]

File: t/torchvision/torchvision_ubi_9.3.sh (hunk 4 above)
Comment (Shafi-Hussain) [reply t/torchvision/torchvision_ubi_9.3.sh:48]: @shubham-dayma-ibm Is the Travis CI using the wrapper script already?
If not then, I do need to install pip and create sym-link for `python`
The base image (UBI9:9.3) used by Travis has python3 installed but not aliased to python. It needs pip to be installed as well

File: t/torchvision/torchvision_ubi_9.3.sh (hunk 5 above)
Comment (Shafi-Hussain) [reply t/torchvision/torchvision_ubi_9.3.sh:37]: I do need to install python3-devel, transitive packages require python header files which isn't there in the base image installation of python

**Comment by Shafi-Hussain** [comment]
Duplicate

**Pull request closed without merging**